pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -412,6 +412,14 @@ func (os *OpenStack) DiskIsAttached(instanceID, volumeID string) (bool, error) {
 		return false, err
 	}
 
+	if instanceID == "" {
+		if volume.AttachedServerId != "" {
+			return true, nil
```

Comment:
If we are sure that if node not found in openstack cinder indicates that volume is already detached. Then I suggest we can the following changes
 In cinder/attacher.go 
```
if err == cloudprovider.InstanceNotFound {
    //return false with nil error
   ...
}
```
You might have this change in the very first version. Sorry for the confusion. 
@rootfs mentioned that cinder might have a more complicated status, and node not exist might not represent volume is already detached successfully from the node. If this is true, we might not return false directly. But I could not think a good way to check it. Passing an empty string for node instance seems not right because our goal is to check whether the volume is still attached to that specified node (not any other node). What does it mean if node id is empty for this function?